File: C++/9. Palindrome Number.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
 public:
  bool isPalindrome(int x) {
    if (x < 0) {
      return false;
    }

    vector<int> v;
    while (x > 0) {
      v.push_back(x % 10);
      x /= 10;
    }

    bool isPalin = true;
    int len = v.size();
    for (int i = 0; i < len / 2; i++) {
      if (v[i] != v[len - i - 1]) {
        isPalin = false;
        break;
      }
    }
    return isPalin;
  }
};

File: JavaScript/9. Palindrome Number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
  if (x < 0) {
    return false;
  }

  const strForm = x.toString();

  const len = strForm.length;
  let isPalin = true;
  for (let i = 0; i < Math.floor(len / 2); i++) {
    if (strForm[i] !== strForm[len - i - 1]) {
      isPalin = false;
      break;
    }
  }
  return isPalin;
};
